### PR TITLE
Support IoT.js

### PR DESCRIPTION
**Pull request by hs0225**
Add IoT.js: Samsung/iotjs#1189

Signed-off-by: Hosung Kim hs852.kim@samsung.com

### CODE REVIEW

File: external/iotjs/src/platform/tizenrt/iotjs_main.c [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+   return 0;
+ }
+ 
+ #ifdef CONFIG_BUILD_KERNEL
+ int main(int argc, FAR char *argv[])
+ #else
+ int iotjs_main(int argc, char *argv[])
+ #endif
+ {
+   return iotjs_register_cmds();
```
Comment (sunghan-chang) [inline external/iotjs/src/platform/tizenrt/iotjs_main.c]: Why do you make two functions to register a TASH command?
In here and In tash_register_cmds.

File: os/Makefile.unix [hunk 2]
```diff
@@ -138,6 +138,15 @@ IOTIVITY_LIBS_DIR=$(EXTDIR)/iotivity/iotivity_$(IOTIVITY_RELEASE)/out/tizenrt/$(
 IOTIVITY_CONFIG_DIR=$(EXTDIR)/iotivity/iotivity_$(IOTIVITY_RELEASE)/resource/c_common
 endif
 
+ifeq ($(CONFIG_ENABLE_IOTJS),y)
```
Comment (sunghan-chang) [inline os/Makefile.unix]: Why do you set it in here, not external?

**Comment by hs0225** [comment]
@sunghan-chang I updated my PR. Please review again.

**Comment by sunghan-chang** [comment]
@hs0225 setjmp is not solved.

**Comment by sunghan-chang** [comment]
@hs0225 After merging #462 , I will merge this too.

**Comment by hs0225** [comment]
@sunghan-chang I see. Thank you.

**Comment by tadziopazur** [comment]
I am not sure I understand this change.
Let me use this analogy - are you trying to put node.js code (and all of its dependencies) into linux kernel tree?

Is any of them going to be developed independently of the other?
Are you going to update all the 3000+ files with every update of the upstream project (iot.js, libtuv, jerry, http-parser)?
Are you trying to glue the TizenRT and IoT.js together? Why? Why like this -- I think there are better ways to do this.

I just cannot comprehend what and why are you trying to do here nor why, it's far beyond me.

I strongly oppose this change.

**Comment by hs0225** [comment]
@tosyu @tadziopazur This PR is the first commit to include IoT.js in TizenRT. When TizenRT is released, IoT.js will be released with a stable version as well. I will not make these commits whenever IoT.js is updated. I will only regularly update the stable version or when TizenRT is released. Separately, we will also provide a way to build the latest IoT.js and TizenRT.

**Comment by tosyu** [comment]
@hs0225 in my opinion a submodule fixed on a specific tag/commit should be used, not a snapshot of the whole tree

**Pull request merged**